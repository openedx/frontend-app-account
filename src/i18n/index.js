// Placeholder be overridden by `make pull_translations`
export default {
  ar: {},
  'zh-hk': {},
  'zh-cn': {},
  uk: {},
  'tr-tr': {},
  th: {},
  te: {},
  ru: {},
  'pt-pt': {},
  'pt-br': {},
  'it-it': {},
  id: {},
  hi: {},
  he: {},
  'fr-ca': {},
  fa: {},
  'es-es': {},
  'es-419': {},
  el: {},
  'de-de': {},
  da: {},
  bo: {},
};
