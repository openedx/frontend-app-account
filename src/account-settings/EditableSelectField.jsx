import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import {
  Button, Form, StatefulButton,
} from '@openedx/paragon';
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import SwitchContent from './SwitchContent';
import messages from './AccountSettingsPage.messages';

import {
  openForm,
  closeForm,
} from './data/actions';
import { editableFieldSelector } from './data/selectors';
import CertificatePreference from './certificate-preference/CertificatePreference';

const EditableSelectField = (props) => {
  const {
    name,
    label,
    emptyLabel,
    type,
    value,
    userSuppliedValue,
    options,
    saveState,
    error,
    confirmationMessageDefinition,
    confirmationValue,
    helpText,
    onEdit,
    onCancel,
    onSubmit,
    onChange,
    isEditing,
    isEditable,
    isGrayedOut,
    intl,
    ...others
  } = props;
  const id = `field-${name}`;

  const handleSubmit = (e) => {
    e.preventDefault();
    onSubmit(name, new FormData(e.target).get(name));
  };

  const handleChange = (e) => {
    onChange(name, e.target.value);
  };

  const handleEdit = () => {
    onEdit(name);
  };

  const handleCancel = () => {
    onCancel(name);
  };

  const renderEmptyLabel = () => {
    if (isEditable) {
      return <Button variant="link" onClick={handleEdit} className="p-0">{emptyLabel}</Button>;
    }
    return <span className="text-muted">{emptyLabel}</span>;
  };

  const renderValue = (rawValue) => {
    if (!rawValue) {
      return renderEmptyLabel();
    }
    let finalValue = rawValue;

    if (options) {
      // Use == instead of === to prevent issues when HTML casts numbers as strings
      // eslint-disable-next-line eqeqeq
      const selectedOption = options.find(option => option.value == rawValue);
      if (selectedOption) {
        finalValue = selectedOption.label;
      }
    }

    if (userSuppliedValue) {
      finalValue += `: ${userSuppliedValue}`;
    }

    return finalValue;
  };

  const renderConfirmationMessage = () => {
    if (!confirmationMessageDefinition || !confirmationValue) {
      return null;
    }
    return intl.formatMessage(confirmationMessageDefinition, {
      value: confirmationValue,
    });
  };
  const selectOptions = options.map((option) => {
    if (option.group) {
      // If the option has a 'group' property, it represents an element with sub-options.
      return (
        <optgroup label={option.label} key={option.label}>
          {option.group.map((subOption) => (
            <option
              value={subOption.value}
              key={`${subOption.value}-${subOption.label}`}
              disabled={subOption?.disabled}
            >
              {subOption.label}
            </option>
          ))}
        </optgroup>
      );
    }
    return (
      <option value={option.value} key={`${option.value}-${option.label}`} disabled={option?.disabled}>
        {option.label}
      </option>
    );
  });

  return (
    <SwitchContent
      expression={isEditing ? 'editing' : 'default'}
      cases={{
        editing: (
          <>
            <form onSubmit={handleSubmit}>
              <Form.Group
                controlId={id}
                isInvalid={error != null}
              >
                <Form.Label size="sm" className="h6 d-block" htmlFor={id}>{label}</Form.Label>
                <Form.Control
                  data-hj-suppress
                  name={name}
                  id={id}
                  type={type}
                  as={type}
                  value={value}
                  onChange={handleChange}
                  {...others}
                >
                  {options.length > 0 && selectOptions}
                </Form.Control>
                {!!helpText && <Form.Text>{helpText}</Form.Text>}
                {error != null && <Form.Control.Feedback>{error}</Form.Control.Feedback>}
                {others.children}
              </Form.Group>
              <p>
                <StatefulButton
                  type="submit"
                  className="mr-2"
                  state={saveState}
                  labels={{
                    default: intl.formatMessage(messages['account.settings.editable.field.action.save']),
                  }}
                  onClick={(e) => {
                    // Swallow clicks if the state is pending.
                    // We do this instead of disabling the button to prevent
                    // it from losing focus (disabled elements cannot have focus).
                    // Disabling it would causes upstream issues in focus management.
                    // Swallowing the onSubmit event on the form would be better, but
                    // we would have to add that logic for every field given our
                    // current structure of the application.
                    if (saveState === 'pending') { e.preventDefault(); }
                  }}
                  disabledStates={[]}
                />
                <Button
                  variant="outline-primary"
                  onClick={handleCancel}
                >
                  {intl.formatMessage(messages['account.settings.editable.field.action.cancel'])}
                </Button>
              </p>
            </form>
            {['name', 'verified_name'].includes(name) && <CertificatePreference fieldName={name} />}
          </>
        ),
        default: (
          <div className="form-group">
            <div className="d-flex align-items-start">
              <h6 aria-level="3">{label}</h6>
              {isEditable ? (
                <Button variant="link" onClick={handleEdit} className="ml-3">
                  <FontAwesomeIcon className="mr-1" icon={faPencilAlt} />{intl.formatMessage(messages['account.settings.editable.field.action.edit'])}
                </Button>
              ) : null}
            </div>
            <p data-hj-suppress className={isGrayedOut ? 'grayed-out' : null}>{renderValue(value)}</p>
            <p className="small text-muted mt-n2">{renderConfirmationMessage() || helpText}</p>
          </div>
        ),
      }}
    />
  );
};

EditableSelectField.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.node]),
  emptyLabel: PropTypes.node,
  type: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  userSuppliedValue: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  })),
  saveState: PropTypes.oneOf(['default', 'pending', 'complete', 'error']),
  error: PropTypes.string,
  confirmationMessageDefinition: PropTypes.shape({
    id: PropTypes.string.isRequired,
    defaultMessage: PropTypes.string.isRequired,
    description: PropTypes.string,
  }),
  confirmationValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  helpText: PropTypes.node,
  onEdit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  isEditing: PropTypes.bool,
  isEditable: PropTypes.bool,
  isGrayedOut: PropTypes.bool,
  intl: intlShape.isRequired,
};

EditableSelectField.defaultProps = {
  value: undefined,
  options: [],
  saveState: undefined,
  label: undefined,
  emptyLabel: undefined,
  error: undefined,
  confirmationMessageDefinition: undefined,
  confirmationValue: undefined,
  helpText: undefined,
  isEditing: false,
  isEditable: true,
  isGrayedOut: false,
  userSuppliedValue: undefined,
};

export default connect(editableFieldSelector, {
  onEdit: openForm,
  onCancel: closeForm,
})(injectIntl(EditableSelectField));
