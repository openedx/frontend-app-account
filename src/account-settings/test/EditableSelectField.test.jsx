import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { IntlProvider } from '@edx/frontend-platform/i18n';

import EditableSelectField from '../EditableSelectField';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}));

jest.mock('@edx/frontend-platform/auth');
jest.mock('../data/selectors', () => jest.fn().mockImplementation(() => ({ certPreferenceSelector: () => ({}) })));

const mockStore = configureStore();

describe('EditableSelectField', () => {
  let props = {};
  let store = {};

  const reduxWrapper = children => (
    <Router>
      <IntlProvider locale="en">
        <Provider store={store}>{children}</Provider>
      </IntlProvider>
    </Router>
  );

  beforeEach(() => {
    store = mockStore();
    props = {
      name: 'testField',
      label: 'Main Label',
      emptyLabel: 'Empty Main Label',
      type: 'text',
      value: 'Test Field',
      userSuppliedValue: '',
      options: [
        {
          label: 'Default Option',
          value: 'defaultOption',
        },
        {
          label: 'User Options',
          group: [
            {
              label: 'Suboption 1',
              value: 'suboption1',
            },
          ],
        },
        {
          label: 'Other Options',
          group: [
            {
              label: 'Suboption 2',
              value: 'suboption2',
            },
            {
              label: 'Suboption 3',
              value: 'suboption3',
            },
          ],
        },
      ],
      saveState: 'default',
      error: '',
      confirmationMessageDefinition: {
        id: 'confirmationMessageId',
        defaultMessage: 'Default Confirmation Message',
        description: 'Description of the confirmation message',
      },
      confirmationValue: 'Confirmation Value',
      helpText: 'Helpful Text',
      isEditing: false,
      isEditable: true,
      isGrayedOut: false,
    };
  });

  afterEach(() => jest.clearAllMocks());

  it('renders EditableSelectField correctly with editing disabled', () => {
    const tree = renderer.create(reduxWrapper(<EditableSelectField {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders EditableSelectField correctly with editing enabled', () => {
    props = {
      ...props,
      isEditing: true,
    };

    const tree = renderer.create(reduxWrapper(<EditableSelectField {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders EditableSelectField with an error', () => {
    const errorProps = {
      ...props,
      error: 'This is an error message',
    };
    const tree = renderer.create(reduxWrapper(<EditableSelectField {...errorProps} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders selectOptions when option has a group', () => {
    const propsWithGroup = {
      ...props,
      options: [
        {
          label: 'User Options',
          group: [
            {
              label: 'Suboption 1',
              value: 'suboption1',
            },
          ],
        },
      ],
    };
    const tree = renderer.create(reduxWrapper(<EditableSelectField {...propsWithGroup} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders selectOptions when option does not have a group', () => {
    const propsWithoutGroup = {
      ...props,
      options: [
        {
          label: 'Default Option',
          value: 'defaultOption',
        },
      ],
    };
    const tree = renderer.create(reduxWrapper(<EditableSelectField {...propsWithoutGroup} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders selectOptions with multiple groups', () => {
    const propsWithGroups = {
      ...props,
      options: [
        {
          label: 'Mixed Options',
          group: [
            {
              label: 'Suboption 1',
              value: 'suboption1',
            },
            {
              label: 'Suboption 2',
              value: 'suboption2',
            },
          ],
        },
      ],
    };
    const tree = renderer.create(reduxWrapper(<EditableSelectField {...propsWithGroups} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
