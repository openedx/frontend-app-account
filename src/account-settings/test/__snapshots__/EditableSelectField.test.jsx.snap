// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableSelectField renders EditableSelectField correctly with editing disabled 1`] = `
<div
  className="pgn-transition-replace-group position-relative"
  style={
    {
      "height": null,
    }
  }
>
  <div
    style={
      {
        "padding": ".1px 0",
      }
    }
  >
    <div
      className="form-group"
    >
      <div
        className="d-flex align-items-start"
      >
        <h6
          aria-level="3"
        >
          Main Label
        </h6>
        <button
          className="ml-3 btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-pencil mr-1"
            data-icon="pencil"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M410.3 231l11.3-11.3-33.9-33.9-62.1-62.1L291.7 89.8l-11.3 11.3-22.6 22.6L58.6 322.9c-10.4 10.4-18 23.3-22.2 37.4L1 480.7c-2.5 8.4-.2 17.5 6.1 23.7s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L387.7 253.7 410.3 231zM160 399.4l-9.1 22.7c-4 3.1-8.5 5.4-13.3 6.9L59.4 452l23-78.1c1.4-4.9 3.8-9.4 6.9-13.3l22.7-9.1 0 32c0 8.8 7.2 16 16 16l32 0zM362.7 18.7L348.3 33.2 325.7 55.8 314.3 67.1l33.9 33.9 62.1 62.1 33.9 33.9 11.3-11.3 22.6-22.6 14.5-14.5c25-25 25-65.5 0-90.5L453.3 18.7c-25-25-65.5-25-90.5 0zm-47.4 168l-144 144c-6.2 6.2-16.4 6.2-22.6 0s-6.2-16.4 0-22.6l144-144c6.2-6.2 16.4-6.2 22.6 0s6.2 16.4 0 22.6z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Edit
        </button>
      </div>
      <p
        className={null}
        data-hj-suppress={true}
      >
        Test Field
      </p>
      <p
        className="small text-muted mt-n2"
      >
        Default Confirmation Message
      </p>
    </div>
  </div>
</div>
`;

exports[`EditableSelectField renders EditableSelectField correctly with editing enabled 1`] = `
<div
  className="pgn-transition-replace-group position-relative"
  style={
    {
      "height": null,
    }
  }
>
  <div
    style={
      {
        "padding": ".1px 0",
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="pgn__form-group"
      >
        <label
          className="pgn__form-label h6 d-block"
          htmlFor="field-testField"
          size="sm"
        >
          Main Label
        </label>
        <div
          className="pgn__form-control-decorator-group"
        >
          <text
            aria-describedby="field-testField-2"
            className="has-value form-control is-invalid"
            data-hj-suppress={true}
            id="field-testField"
            name="testField"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="Test Field"
          >
            <option
              value="defaultOption"
            >
              Default Option
            </option>
            <optgroup
              label="User Options"
            >
              <option
                value="suboption1"
              >
                Suboption 1
              </option>
            </optgroup>
            <optgroup
              label="Other Options"
            >
              <option
                value="suboption2"
              >
                Suboption 2
              </option>
              <option
                value="suboption3"
              >
                Suboption 3
              </option>
            </optgroup>
          </text>
        </div>
        <div
          className="pgn__form-text pgn__form-text-default"
        >
          <div>
            Helpful Text
          </div>
        </div>
        <div
          className="pgn__form-control-description pgn__form-text pgn__form-text-invalid"
          id="field-testField-2"
        >
          <span
            className="pgn__icon"
          >
            <svg
              aria-hidden={true}
              fill="none"
              focusable={false}
              height={24}
              role="img"
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <div />
        </div>
      </div>
      <p>
        <button
          aria-disabled={false}
          aria-live="assertive"
          className="pgn__stateful-btn pgn__stateful-btn-state-default mr-2 btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span
            className="d-flex align-items-center justify-content-center"
          >
            <span>
              Save
            </span>
          </span>
        </button>
        <button
          className="btn btn-outline-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </p>
    </form>
  </div>
</div>
`;

exports[`EditableSelectField renders EditableSelectField with an error 1`] = `
<div
  className="pgn-transition-replace-group position-relative"
  style={
    {
      "height": null,
    }
  }
>
  <div
    style={
      {
        "padding": ".1px 0",
      }
    }
  >
    <div
      className="form-group"
    >
      <div
        className="d-flex align-items-start"
      >
        <h6
          aria-level="3"
        >
          Main Label
        </h6>
        <button
          className="ml-3 btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-pencil mr-1"
            data-icon="pencil"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M410.3 231l11.3-11.3-33.9-33.9-62.1-62.1L291.7 89.8l-11.3 11.3-22.6 22.6L58.6 322.9c-10.4 10.4-18 23.3-22.2 37.4L1 480.7c-2.5 8.4-.2 17.5 6.1 23.7s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L387.7 253.7 410.3 231zM160 399.4l-9.1 22.7c-4 3.1-8.5 5.4-13.3 6.9L59.4 452l23-78.1c1.4-4.9 3.8-9.4 6.9-13.3l22.7-9.1 0 32c0 8.8 7.2 16 16 16l32 0zM362.7 18.7L348.3 33.2 325.7 55.8 314.3 67.1l33.9 33.9 62.1 62.1 33.9 33.9 11.3-11.3 22.6-22.6 14.5-14.5c25-25 25-65.5 0-90.5L453.3 18.7c-25-25-65.5-25-90.5 0zm-47.4 168l-144 144c-6.2 6.2-16.4 6.2-22.6 0s-6.2-16.4 0-22.6l144-144c6.2-6.2 16.4-6.2 22.6 0s6.2 16.4 0 22.6z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Edit
        </button>
      </div>
      <p
        className={null}
        data-hj-suppress={true}
      >
        Test Field
      </p>
      <p
        className="small text-muted mt-n2"
      >
        Default Confirmation Message
      </p>
    </div>
  </div>
</div>
`;

exports[`EditableSelectField renders selectOptions when option does not have a group 1`] = `
<div
  className="pgn-transition-replace-group position-relative"
  style={
    {
      "height": null,
    }
  }
>
  <div
    style={
      {
        "padding": ".1px 0",
      }
    }
  >
    <div
      className="form-group"
    >
      <div
        className="d-flex align-items-start"
      >
        <h6
          aria-level="3"
        >
          Main Label
        </h6>
        <button
          className="ml-3 btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-pencil mr-1"
            data-icon="pencil"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M410.3 231l11.3-11.3-33.9-33.9-62.1-62.1L291.7 89.8l-11.3 11.3-22.6 22.6L58.6 322.9c-10.4 10.4-18 23.3-22.2 37.4L1 480.7c-2.5 8.4-.2 17.5 6.1 23.7s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L387.7 253.7 410.3 231zM160 399.4l-9.1 22.7c-4 3.1-8.5 5.4-13.3 6.9L59.4 452l23-78.1c1.4-4.9 3.8-9.4 6.9-13.3l22.7-9.1 0 32c0 8.8 7.2 16 16 16l32 0zM362.7 18.7L348.3 33.2 325.7 55.8 314.3 67.1l33.9 33.9 62.1 62.1 33.9 33.9 11.3-11.3 22.6-22.6 14.5-14.5c25-25 25-65.5 0-90.5L453.3 18.7c-25-25-65.5-25-90.5 0zm-47.4 168l-144 144c-6.2 6.2-16.4 6.2-22.6 0s-6.2-16.4 0-22.6l144-144c6.2-6.2 16.4-6.2 22.6 0s6.2 16.4 0 22.6z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Edit
        </button>
      </div>
      <p
        className={null}
        data-hj-suppress={true}
      >
        Test Field
      </p>
      <p
        className="small text-muted mt-n2"
      >
        Default Confirmation Message
      </p>
    </div>
  </div>
</div>
`;

exports[`EditableSelectField renders selectOptions when option has a group 1`] = `
<div
  className="pgn-transition-replace-group position-relative"
  style={
    {
      "height": null,
    }
  }
>
  <div
    style={
      {
        "padding": ".1px 0",
      }
    }
  >
    <div
      className="form-group"
    >
      <div
        className="d-flex align-items-start"
      >
        <h6
          aria-level="3"
        >
          Main Label
        </h6>
        <button
          className="ml-3 btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-pencil mr-1"
            data-icon="pencil"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M410.3 231l11.3-11.3-33.9-33.9-62.1-62.1L291.7 89.8l-11.3 11.3-22.6 22.6L58.6 322.9c-10.4 10.4-18 23.3-22.2 37.4L1 480.7c-2.5 8.4-.2 17.5 6.1 23.7s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L387.7 253.7 410.3 231zM160 399.4l-9.1 22.7c-4 3.1-8.5 5.4-13.3 6.9L59.4 452l23-78.1c1.4-4.9 3.8-9.4 6.9-13.3l22.7-9.1 0 32c0 8.8 7.2 16 16 16l32 0zM362.7 18.7L348.3 33.2 325.7 55.8 314.3 67.1l33.9 33.9 62.1 62.1 33.9 33.9 11.3-11.3 22.6-22.6 14.5-14.5c25-25 25-65.5 0-90.5L453.3 18.7c-25-25-65.5-25-90.5 0zm-47.4 168l-144 144c-6.2 6.2-16.4 6.2-22.6 0s-6.2-16.4 0-22.6l144-144c6.2-6.2 16.4-6.2 22.6 0s6.2 16.4 0 22.6z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Edit
        </button>
      </div>
      <p
        className={null}
        data-hj-suppress={true}
      >
        Test Field
      </p>
      <p
        className="small text-muted mt-n2"
      >
        Default Confirmation Message
      </p>
    </div>
  </div>
</div>
`;

exports[`EditableSelectField renders selectOptions with multiple groups 1`] = `
<div
  className="pgn-transition-replace-group position-relative"
  style={
    {
      "height": null,
    }
  }
>
  <div
    style={
      {
        "padding": ".1px 0",
      }
    }
  >
    <div
      className="form-group"
    >
      <div
        className="d-flex align-items-start"
      >
        <h6
          aria-level="3"
        >
          Main Label
        </h6>
        <button
          className="ml-3 btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-pencil mr-1"
            data-icon="pencil"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M410.3 231l11.3-11.3-33.9-33.9-62.1-62.1L291.7 89.8l-11.3 11.3-22.6 22.6L58.6 322.9c-10.4 10.4-18 23.3-22.2 37.4L1 480.7c-2.5 8.4-.2 17.5 6.1 23.7s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L387.7 253.7 410.3 231zM160 399.4l-9.1 22.7c-4 3.1-8.5 5.4-13.3 6.9L59.4 452l23-78.1c1.4-4.9 3.8-9.4 6.9-13.3l22.7-9.1 0 32c0 8.8 7.2 16 16 16l32 0zM362.7 18.7L348.3 33.2 325.7 55.8 314.3 67.1l33.9 33.9 62.1 62.1 33.9 33.9 11.3-11.3 22.6-22.6 14.5-14.5c25-25 25-65.5 0-90.5L453.3 18.7c-25-25-65.5-25-90.5 0zm-47.4 168l-144 144c-6.2 6.2-16.4 6.2-22.6 0s-6.2-16.4 0-22.6l144-144c6.2-6.2 16.4-6.2 22.6 0s6.2 16.4 0 22.6z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          Edit
        </button>
      </div>
      <p
        className={null}
        data-hj-suppress={true}
      >
        Test Field
      </p>
      <p
        className="small text-muted mt-n2"
      >
        Default Confirmation Message
      </p>
    </div>
  </div>
</div>
`;
